src/home/components/ResetPasswordContents/EmailVerification/useEmailVerification.ts

```diff
@@ -0,0 +1,77 @@
+import { useState, useEffect } from 'react';
+import { useSecondTimer } from '@/hooks/useSecondTimer';
+import { getAuthVerificationCheck, postAuthVerificationEmail } from '@/home/apis/authVerification';
+import { useFullEmail } from '@/hooks/useFullEmail';
+import { STORAGE_KEYS } from '@/constants/storage.constant';
+
+interface UseEmailVerificationProps {
+  email: string;
+  onConfirm: () => void;
+}
+
+export const useEmailVerification = ({ email, onConfirm }: UseEmailVerificationProps) => {
+  const { leftTime, resetTimer } = useSecondTimer(480);
+  const fullEmail = useFullEmail(email);
+  const [error, setError] = useState<string | null>(null);
+  const [isResending, setIsResending] = useState(false);
+
+  useEffect(() => {
+    const session = sessionStorage.getItem(STORAGE_KEYS.EMAIL_AUTH_SESSION_TOKEN);
+    if (session) {
+      verifyEmailSession(session, true);
+    }
+  }, []);
+
+  const verifyEmailSession = async (session: string, isInitialCheck: boolean = false) => {
+    const { data, error } = await getAuthVerificationCheck({ session });
+    if (data?.isVerified) {
+      setError(null);
+      onConfirm();
+    } else if (!isInitialCheck || error) {
+      setError('이메일 인증을 완료해주세요.');
+    }
+  };
+
+  const handleVerification = async () => {
+    const session = sessionStorage.getItem(STORAGE_KEYS.EMAIL_AUTH_SESSION_TOKEN);
+    const sessionExpiresAt = sessionStorage.getItem(
+      STORAGE_KEYS.EMAIL_AUTH_SESSION_TOKEN_EXPIRED_IN
+    );
+
+    if (!session || !sessionExpiresAt) {
+      setError('이메일 인증을 완료해주세요.');
+      return;
+    }
+    await verifyEmailSession(session);
+  };
+
+  const handleTimer = (seconds: number) => {
+    const minutes = Math.floor(seconds / 60);
+    const secs = seconds % 60;
+    return `${minutes.toString().padStart(2, '0')}:${secs.toString().padStart(2, '0')}`;
+  };
+
+  const handleResendEmail = async () => {
+    setIsResending(true);
+    resetTimer();
+
+    const response = await postAuthVerificationEmail({
+      email: fullEmail,
+      verificationType: 'PASSWORD',
+    });
+
+    if (response.error) {
+      setError('이메일 인증을 완료해주세요.');
+    }
+    setIsResending(false);
+  };
```

Comment:
이거 제롬이 만든 `usePreventDuplicateClick` 써도 될듯합니다!
loading 관련된 상태의 변경을 가릴 수 있어서 쓰면 기능 분할도 되고 좋을듯합니다!

```suggestion
const { disabled, handleClick } = usePreventDuplicateClick();

const handleResendEmail = async () => {
    resetTimer();

    const response = await postAuthVerificationEmail({
      email: fullEmail,
      verificationType: 'PASSWORD',
    });

    if (response.error) {
      setError('이메일 인증을 완료해주세요.');
    }
};

return {
  ...,
  handleResendEmail: handleClick(handleResendEmail),
  isResending: disabled,
}
```

Reply:
확실히 저도 타입 범용성 문제 때문에 안쓴 코드도 있습니다!
그대로 진행시키시져